x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotIndexInput.java

```diff
@@ -41,20 +44,31 @@
     private final long length;
 
     private long position;
-    private boolean closed;
+    private volatile boolean closed;
 
-    public SearchableSnapshotIndexInput(final BlobContainer blobContainer, final FileInfo fileInfo) {
-        this("SearchableSnapshotIndexInput(" + fileInfo.physicalName() + ")", blobContainer, fileInfo, 0L, 0L, fileInfo.length());
+    // optimisation for the case where we perform a single seek, then read a large block of data sequentially, then close the input
+    @Nullable // if not currently reading sequentially
+    private StreamForSequentialReads streamForSequentialReads;
```

Comment:
Maybe update the class javadoc to explain how/why we use this?

Reply:
sure, done in f18251a597595b441b09ac46432623163a25caf0